Yoast/javascript
# Create TwitterImage component

## Summary

This PR can be summarized in the following changelog entry:

* Adds a TwitterImage component

## Relevant technical choices:

* We do not want images to get warped if their dimensions do not match Twitter's image dimensions. Therefore, I cropped the images such that the dimension (width/height) which 'overflows' Twitter's image dimensions, is cut off equally on both sides.
* If no image is provided, a placeholder image is shown. 
* The image that is uploaded must meet several requirements (e.g., it should't be too big, see #50). However, these requirements and corresponding checks have not yet been implemented; this should happen in a new, future issue. I didn't change or remove the code which creates an `ErrorImage` (it is copied from the FacebookImage component), this will happen in #127.
* I made some changes to `TwitterPreview.js`, `TwitterPreviewExample.js` and `TwitterPreviewTest.js`, in order to obtain relevant previews and prevent the tests from failing. However, please note that the TwitterPreview component falls outside the scope of the current issue.
* I refactored the code so that `FacebookImage.js` and `TwitterImage.js` now share the following helper files: `determineImageProperties.js`, `delayComponentsSnapshot.js` and `mockImage.js`.
* `mockImage.js` was created in order to test the `determineImageProperties` function from `determineImageProperties.js`. This was necessary because `img.onload` only works in the browser, but not in Jest. Therefore, we mock the image. Currently, I only use the `createSuccessfulImage` function, because everything with erroring images should be dealt with in other issues? Then, we can  make use of the `createFailingImage` function.
* `mockImage.js` is not covered in unit tests because that seemed overkill (`mockImage.js` itself was created to make other unit tests work). `TwitterImage.js` is not at 100% coverage because the missing tests concern error states, which will be removed later (after this is done, coverage should be at 100%).

## Test instructions

This PR can be tested by following these steps:

* Check out this branch.
* From the `javascript/apps/components` folder in the terminal, run `yarn` and `yarn start`.
* `yarn test` should not result in any errors.
* Start the stand-alone by navigating to `http://localhost:3333/` in the browser.
* Go to the `TwitterPreview` page by clicking on the button in the top right.
* Check if the images look alright.
    * If the image is described as "large", it will later appear in a `Summary with large image` card, and should be rendered as a landscape with a 2:1 (width:height) ratio.
    * If the card type is described as "small", it will later appear in a `Summary` card, and should be rendered as a square.

## UI changes

## Quality assurance


Fixes #50
